### ISSUE TITLE
Constant stream of exceptions when starting Retopo in Edit mode.

### ISSUE THREAD

**Issue by VincentKueszter** [CGCookie/retopoflow#766]
Environment:

    RetopoFlow: 3.0.0α
    Blender: 2.81.16 master 2019-11-20

Issue:

This might be related to #763, but I don't have just one error, but a constant steam of silent errors in the background, which are different from the one mentioned in #763.
This blocks me from enabling RetopoFlow Mode, since I only can enable that in Edit Mode.
(When trying it in Object Mode, I can only select "RF: Create new target", no matter which object is selected. I don't know if that is intended.)

Anonymized excerpt of the exceptions:
```
RetopoFlow: self.update_displace_option()
CookieCutter handling exception caught while trying to "call start()"
____________________________________________________________________________________________________
Exception Info
- Time: 2019-11-29 18:54:44.916652
- Count: 1
- Hash: 00e8350713b7282e149fd3e36a510001
- Info:
  - EXCEPTION (<class 'AttributeError'>): rot_object
  -     C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py
  - 000 0688:__delattr__() return super().__delattr__(attr)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_blender.py
  - 001 0190:end_rotate_about_active() del self.rot_object
  -     .../common\decorators.py
  - 002 0188:callit() ret = fn(*args, **kwargs)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_blender.py
  - 003 0172:setup_rotate_about_active() self.end_rotate_about_active()      # clear out previous rotate-about object
  -     .../common\decorators.py
  - 004 0188:callit() ret = fn(*args, **kwargs)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\retopoflow.py
  - 005 0082:start() self.setup_rotate_about_active()
  -     .../cookiecutter\cookiecutter.py
  - 006 0092:invoke() try: self.start()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
CookieCutter_Exceptions._callback_exception_callbacks: rot_object
CookieCutter caught exception while calling back exception callbacks: handle_exception
____________________________________________________________________________________________________
Exception Info
- Time: 2019-11-29 18:54:44.927702
- Count: 2
- Hash: edfd626a5bcb51585296152602650c08
- Info:
  - EXCEPTION (<class 'AttributeError'>): 'VIEW3D_OT_RetopoFlow' object has no attribute 'undo'
  -     C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py
  - 000 0674:__getattribute__() return super().__getattribute__(attr)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_undo.py
  - 001 0089:undo_stack_actions() return [u['action'] for u in reversed(self.undo)]
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_ui.py
  - 002 0273:alert_user() get_trace_details(self.undo_stack_actions(), msghash=msghash, message=message_orig),
  - 003 0131:handle_exception() self.alert_user(title='Exception caught', message=message, level='exception', msghash=h)
  -     .../cookiecutter\cookiecutter_exceptions.py
  - 004 0043:_callback_exception_callbacks() fn(e)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
CookieCutter handling exception caught while trying to "draw pre3d"
____________________________________________________________________________________________________
Exception Info
- Time: 2019-11-29 18:54:44.933703
- Count: 3
- Hash: 196bc73dafbb20dc500188e19876e7b1
- Info:
  - EXCEPTION (<class 'AttributeError'>): 'VIEW3D_OT_RetopoFlow' object has no attribute 'rftool'
  -     C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py
  - 000 0674:__getattribute__() return super().__getattribute__(attr)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_states.py
  - 001 0043:update() self.rftool._callback('target change')
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_drawing.py
  - 002 0049:predraw() self.update(timer=False)
  -     .../common\drawing.py
  - 003 1199:_call() for fn in self._fns[n]: fn(self.obj)
  - 004 1203:_pre() self._call('pre')
  - 005 1206:pre3d() self._pre()
  -     .../cookiecutter\cookiecutter_ui.py
  - 006 0089:preview() try: self.drawcallbacks.pre3d()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
CookieCutter_Exceptions._callback_exception_callbacks: 'VIEW3D_OT_RetopoFlow' object has no attribute 'rftool'
CookieCutter caught exception while calling back exception callbacks: handle_exception
____________________________________________________________________________________________________
Exception Info
- Time: 2019-11-29 18:54:44.939705
- Count: 4
- Hash: edfd626a5bcb51585296152602650c08
- Info:
  - EXCEPTION (<class 'AttributeError'>): 'VIEW3D_OT_RetopoFlow' object has no attribute 'undo'
  -     C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py
  - 000 0674:__getattribute__() return super().__getattribute__(attr)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_undo.py
  - 001 0089:undo_stack_actions() return [u['action'] for u in reversed(self.undo)]
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_ui.py
  - 002 0273:alert_user() get_trace_details(self.undo_stack_actions(), msghash=msghash, message=message_orig),
  - 003 0131:handle_exception() self.alert_user(title='Exception caught', message=message, level='exception', msghash=h)
  -     .../cookiecutter\cookiecutter_exceptions.py
  - 004 0043:_callback_exception_callbacks() fn(e)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Traceback (most recent call last):
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\cookiecutter\cookiecutter_ui.py", line 89, in preview
    try: self.drawcallbacks.pre3d()
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\common\drawing.py", line 1206, in pre3d
    self._pre()
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\common\drawing.py", line 1203, in _pre
    self._call('pre')
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\common\drawing.py", line 1199, in _call
    for fn in self._fns[n]: fn(self.obj)
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_drawing.py", line 49, in predraw
    self.update(timer=False)
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_states.py", line 43, in update
    self.rftool._callback('target change')
  File "C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py", line 674, in __getattribute__
    return super().__getattribute__(attr)
AttributeError: 'VIEW3D_OT_RetopoFlow' object has no attribute 'rftool'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\cookiecutter\cookiecutter_ui.py", line 92, in preview
    ScissorStack.end(force=True)
  File "C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\addon_common\common\drawing.py", line 1065, in end
    ScissorStack.stack = ScissorStack.stack[:1]
TypeError: 'NoneType' object is not subscriptable
CookieCutter handling exception caught while trying to "draw post3d"
____________________________________________________________________________________________________
Exception Info
- Time: 2019-11-29 18:54:44.947706
- Count: 5
- Hash: b6351e1f95aa1f3342213198f1432b08
- Info:
  - EXCEPTION (<class 'AttributeError'>): 'VIEW3D_OT_RetopoFlow' object has no attribute 'view_version'
  -     C:\Program Files\Blender Foundation\Blender\2.80\scripts\modules\bpy_types.py
  - 000 0674:__getattribute__() return super().__getattribute__(attr)
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_states.py
  - 001 0049:update() if self.view_version != view_version:
  -     C:\Users\[...]\AppData\Roaming\Blender Foundation\Blender\2.80\scripts\addons\RetopoFlow\retopoflow\rf\rf_drawing.py
  - 002 0049:predraw() self.update(timer=False)
  -     .../common\drawing.py
  - 003 1199:_call() for fn in self._fns[n]: fn(self.obj)
  - 004 1203:_pre() self._call('pre')
  - 005 1209:post3d() self._pre()
  -     .../cookiecutter\cookiecutter_ui.py
  - 006 0094:postview() try: self.drawcallbacks.post3d()
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
By the way:
After some time, the log is peppered with:
```
Caught exception while trying to pull issues from GitHub
URL: "https://api.github.com/repos/CGCookie/retopoflow/issues?state=all"
HTTP Error 403: Forbidden
```

**Comment by vxlcoder** [comment]
Thanks for the report! Will look into it!

**Comment by vxlcoder** [comment]
@VincentKueszter, can you replicate this issue starting from the default scene?  If not, what are your steps to reproducing this error?  Be sure to include a list of objects in the current scene, and whether they are visible, are you currently editing one of them, etc. Alternatively, are you able to share a .blend file in which this error occurs?  I think I might know what is causing this, but I cannot replicate it here.

the log reports about the 403 from GitHub: this is from the error catcher trying to check GH if the error has been reported before.  GH only allows you to check up to 60 times in an hour, and you probably went over that with this particular error.  (the count resets after an hour)

Thanks!

**Comment by VincentKueszter** [comment]
I think it would be easiest if I gave you the blend file, since I played around with the retopo for quite a bit, before noticing it wouldn't load on the next day.
I don't want to share it publicly, how can I send you a link in private?

**Comment by vxlcoder** [comment]
that would be super helpful, @VincentKueszter!  i'll get you an email address that you can send the .blend (or better a dropbox link, etc.) to in just a sec...

**Comment by vxlcoder** [comment]
@VincentKueszter, please send to retopoflow@gmail.com.  thanks!

**Comment by VincentKueszter** [comment]
I sent you the link!

**Comment by VincentKueszter** [comment]
Issue is fixed by c66a488, thanks a lot!

**Comment by vxlcoder** [comment]
thank you for your help!  closing
